### PR TITLE
fix(tools): adding process.env.NODE_ENV replacement to dev-server.ts

### PR DESCRIPTION
**Pull request by brianferry**
## Description

- Adding package @rollup/plugin-replace to pfe-tools. 
- Adding replace function to plugins in dev-server.ts to replace process.env.NODE_ENV with 'production'. 

### Related issues

- (#1941) Issue with process.env.NODE_ENV being undefined inside of components

### Preview

Link(s) to demo page(s) where this element can be viewed:
- [Link](https://deploy-preview-<pr_number>--happy-galileo-ea79c4.netlify.app/examples/) 

### Merging

Please **squash** when merging and ensure your commit message uses [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) formatting.

**Be sure to share your updates with the [patternfly-elements-contribute@redhat.com](mailto:patternfly-elements-contribute@redhat.com) mailing list!**

### CODE REVIEW

**Review by zeroedin — COMMENTED** [review]
LGTM.

File: package.json [hunk 1]
```diff
@@ -71,6 +71,7 @@
     "@commitlint/config-lerna-scopes": "^16.0.0",
     "@patternfly/eslint-config-elements": "^0.0.2-next.1",
     "@patternfly/pfe-tools": "^0.1.0-next.4",
+    "@rollup/plugin-replace": "^4.0.0",
```
Comment (bennypowers) [inline package.json:74]: should be added to pfe-tools package instead of the root

File: .changeset/rotten-cups-roll.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"@patternfly/pfe-tools": minor
+---
+
+Adding @rollup/plugin-replace to dev server to fix issue with undefined process.env.NODE_ENV
```
Comment (bennypowers) [inline .changeset/rotten-cups-roll.md:5]:
```suggestion
Configures dev server to automatically replace instances of `process.env.NODE_ENV` with `production` in library code.
```

File: tools/pfe-tools/dev-server.ts [hunk 3]
```diff
@@ -113,6 +115,10 @@ export function pfeDevServerConfig(options?: PfeDevServerConfigOptions): DevServ
       bindNodeDataToBrowser({ cwd }),
       // load .scss files as lit CSSResult modules
       litcss({ include: ['**/*.scss'], transform: transformSass }),
+      replace({
+        'preventAssignment': true,
+        'process.env.NODE_ENV': JSON.stringify( 'production' )
```
Comment (bennypowers) [inline tools/pfe-tools/dev-server.ts:120]: we will likely also need to do this in esbuild

**Review by bennypowers — CHANGES_REQUESTED** [review]

**Review by bennypowers — COMMENTED** [review]
I tried adding this to `/elements/pfe-accordion/pfe-accordion.ts` 
```js
console.log(process.env.NODE_ENV)
```
then ran 
```bash
npm run build -w @patternfly/pfe-accordion
```

then
```
$  grep console elements/pfe-accordion/pfe-accordion.js
console.log("development");
```

which i do believe should actually be 'production', right?

File: tools/pfe-tools/esbuild.ts [hunk 4]
```diff
@@ -145,6 +145,9 @@ export async function singleFileBuild(options?: PfeEsbuildSingleFileOptions) {
       sourcemap: true,
       treeShaking: true,
       watch: false,
+      define: {
```
Comment (bennypowers) [inline tools/pfe-tools/esbuild.ts:148]: this should be duplicated ~line 219 in `pfeBuild()` as well

File: .changeset/rotten-cups-roll.md [hunk 5]
```diff
@@ -0,0 +1,5 @@
+---
+"@patternfly/pfe-tools": minor
+---
+
+Configures dev server to automatically replace instances of `process.env.NODE_ENV` with `production` in library code.
```
Comment (bennypowers) [inline .changeset/rotten-cups-roll.md:5]:
```suggestion
Automatically replaces `process.env.NODE_ENV` with `production` both in dev server and production bundles.
```

**Review by bennypowers — CHANGES_REQUESTED** [review]

**Review by bennypowers — CHANGES_REQUESTED** [review]

**Comment by brianferry** [comment]
Updated the esbuild.ts for production builds, removed console log & updated changeset.

**Review by bennypowers — APPROVED** [review]

**Pull request merged**